packages/engine/src/getContext.js

```diff
@@ -93,6 +93,26 @@ function getContext({ config, lowdefy, development = false }) {
   _internal.update = () => {
     _internal.RootBlocks.update();
   };
+  _internal.runOnInit = async (progress) => {
+    progress();
+    if (!_internal.State.initialized && !_internal.onInitDone) {
+      await _internal.RootBlocks.areas.root.blocks[0].triggerEvent({
+        name: 'onInit',
+        progress,
+      });
+      _internal.State.freezeState();
+      _internal.onInitDone = true;
+    }
+  };
+  _internal.runOnInitAsync = async (progress) => {
+    if (!_internal.State.initialized && !_internal.onInitAsyncDone) {
```

Comment:
Do we want to check that `onInit` has already run, or do we just trust engine will run these correctly?

Reply:
They are triggered by `MountEvents` as defined in `getContext` so the order is not implemented by engine. I guess we could add `onInitDone === true`.. Not really sure if it will add additional value.